lib/commands/location.js

```diff
@@ -19,6 +19,12 @@ export default {
    * @this {XCUITestDriver}
    */
   async getGeoLocation() {
+    // Currently we proxy the setGeoLocation to mobile:setSimulatedLocation for iOS 17+.
+    // It would be helpful to address to use "mobile:getSimulatedLocation" for iOS 17+.
+    if (this.isRealDevice() && this.opts.platformVersion && util.compareVersions(this.opts.platformVersion, '>=', '17.0')) {
+      this.log.info(`Location set by mobile:setSimulatedLocation method should get via mobile:getSimulatedLocation method.`);
```

Comment:
we may set a flag if one has used `mobile:getSimulatedLocation` API and show it as a warning if the flag is true

Reply:
Could be...? The simulated location is WDA process wide, so in case WDA process keeps running while Appium restarts, the flag could tell a lie, so I haven't added such a flag right now

Reply:
if this is connected to WDA then we may also introduce such feature in WDA itself. E.g. calling '/wda/device/location' returns the usual value if setSimulatedLocation was not called before, otherwise it returns the simulated one

Reply:
Yea. I haven't checked simulatos, but real devices return null values for getSimulatedLocation if none of set the simulated values via WDA. Probably as a follow up pr, here can:

- call getSimulatedLocation first
    - if the return value includes null -> then log this info and explain trying to get the device's actual location info
    - if the return value does not include null -> print this value is by simulated value

Reply:
> call getSimulatedLocation first

probably only for iOS 17+, but otherwise looks good.
AFAIK very small percentage of users check server logs